src/4.3_to_5.0/migrateSettingsFolder.ts

```diff
@@ -76,6 +78,44 @@ function migrateSettingsMap(legacySettingsMap: {
     );
   }
 
+  /* Smart filtering settings start. */
```

Comment:
To split code, it would be more robust to do:

```
/**
 * Adds the smart-filtering settings from `legacySettingsMap` into `migratedSettingsMap`.
 * Mutates `migratedSettingsMap`.
 */
migrateSmartFilteringSettings(migratedSettingsMap, legacySettingsMap);
```
rather than to add these comments, which are likely to become out of place in the future.

Another benefit is that this function would be easily unit testable, with unit tests that will be more maintainable than the one that generates a 3K lines snapshot.